bool flag
bool flag_error
string color_information
int32 error_range
int32 num_judgments
int32 depth
int32 ID
